org-code-javabuilder/theater/src/main/java/org/code/theater/Stage.java

```diff
@@ -362,6 +366,9 @@ public void play() {
       this.gifWriter.close();
       this.audioWriter.writeToAudioStreamAndClose();
 
+      this.writeImageAndAudioToFile();
+
+      // TODO: Remove these messages once we've updated the client to load from file.
       this.outputAdapter.sendMessage(ImageEncoder.encodeStreamToMessage(this.imageOutputStream));
       this.outputAdapter.sendMessage(SoundEncoder.encodeStreamToMessage(this.audioOutputStream));
```

Comment:
Looking at this now: since we're now allowing the client to accept both URLs and base64 encoded files, if we send these back to back, won't we end up overwriting the S3 messages with the encoded file messages?